book/src/integrations/oauth2.md

```diff
@@ -328,14 +329,29 @@ kanidm system oauth2 create-public <name> <displayname> <origin>
 kanidm system oauth2 create-public mywebapp "My Web App" https://webapp.example.com
 ```
 
-To allow localhost redirection
+## Native Applications
+
+Some applications will run a local web server on the user's device  which directs users to the IDP for
+authentication, then back to the local server. [BCP212](https://www.rfc-editor.org/info/bcp212)
+"OAuth 2.0 for Native Apps" specifies the rules for this.
+
+First allow localhost redirects:
 
 ```bash
 kanidm system oauth2 enable-localhost-redirects <name>
 kanidm system oauth2 disable-localhost-redirects <name>
 kanidm system oauth2 enable-localhost-redirects mywebapp
 ```
 
+Then configure a local URL. We're going to match on "is it a loopback address" and "does the path match" so specify something similar to the following:
+
+```bash
```

Comment:
This is a lot of configuration to make this work. It also enshrines magic which is that we have to handle the possible localhost variants, and we set ports etc. There also likely isn't a guarantee the path is stable.

We also need to be clear that these are *only* allowed on public clients which enforce pkce.